src/pybind/mgr/orchestrator.py

```diff
@@ -1030,11 +1037,15 @@ def _orchestrator_wait(self, completions):
 
         :param completions: List of Completions
         :raises NoOrchestrator:
+        :raises RuntimeError: something went wrong while calling the process method.
         :raises ImportError: no `orchestrator_cli` module or backend not found.
         """
         for c in completions:
             self._update_completion_progress(c)
-        while not self.wait(completions):
+        while any(not c.is_complete for c in completions):
+            self.wait(completions)
```

Comment:
Should this be `self.process(completions)`?

Reply:
Right. Fixed one commit later in https://github.com/ceph/ceph/pull/30262/commits/5012a21b91fbcdb0c34622be2f8585ffbf96b10a#diff-dfbc76d00f82320b0e6ac732319aa545R1268

Would it make sense to fix this in this commit, too?

Reply:
*gah* sorry, missed that :-/  If you can be bothered moving the fix to this commit it might be slightly cleaner, but I'm really not too fussed :-)